### PR TITLE
Initial development for a configuration file

### PR DESCRIPTION
**Pull request by caffix**

### CODE REVIEW

File: amass/config.go [hunk 1]
```diff
@@ -186,3 +218,108 @@ func (c *Config) GetAPIKey(source string) *APIKey {
...
 }
+
+// LoadSettings parses settings from an .ini file and assigns them to the Config.
+func (c *Config) LoadSettings(path string) error {
+	cfg, err := ini.LoadSources(ini.LoadOptions{
+		Insensitive:  true,
+		AllowShadows: true,
+	}, path)
+	if err != nil {
+		return err
```
Comment (kalbasit) [inline amass/config.go]: This isn't a blocker for this PR but I just wanted to point that out. I found with Go that errors can sometimes be ambiguous as you don't really know where they were returned. At work, I've opted for doing `return errors.Wrap(err, "reason why I am returning this error")` which shows up as `reason why I am returning this error: whatever the error was`. The errors package can be found at https://github.com/pkg/errors

File: amass/config.go [hunk 2]
```diff
@@ -186,3 +218,108 @@ func (c *Config) GetAPIKey(source string) *APIKey {
...
+	}
+	// Attempt to load a wordlist provided via the configuration file
+	if cfg.Section(ini.DEFAULT_SECTION).HasKey("wordlist_file") {
+		wordlist := cfg.Section(ini.DEFAULT_SECTION).Key("wordlist_file").String()
+
+		if list, err := getWordList(wordlist); err == nil {
+			c.Wordlist = list
+		} else {
+			return err
+		}
```
Comment (kalbasit) [inline amass/config.go]: The if..else is usually frowned upon in Go, the following the preferable:

```
list, err := getWordList(wordlist)
if err != nil {
    return err
}

c.Wordlist = list
```

this is true below as well.

File: amass/config.go (hunk 1 above)
Comment (caffix) [reply amass/config.go]: I agree. I usually do wrap error messages, but rushed last night. Thank you for catching it!

File: amass/config.go (hunk 2 above)
Comment (caffix) [reply amass/config.go]: Fixed those up

**Comment by caffix** [comment]
Great! This should make a good start

**Pull request merged**